nav2_map_server/src/map_3d/map_io_3d.cpp

```diff
@@ -0,0 +1,382 @@
+// Copyright (c) 2020 Shivam Pandey
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*Map-io library for 3D pointcloud implementation*/
+
+#include "nav2_map_server/map_3d/map_io_3d.hpp"
+
+#ifndef _WIN32
+#include <libgen.h>
+#endif
+#include <iostream>
+#include <string>
+#include <vector>
+#include <stdexcept>
+
+#include "yaml-cpp/yaml.h"
+#include "nav2_map_server/map_3d/pcl_helper.hpp"
+
+#include "pcl/io/pcd_io.h"
+#include "Eigen/Core"
+
+#ifdef _WIN32
+// https://github.com/rtv/Stage/blob/master/replace/dirname.c
```

Comment:
Woah, that's a GPL license, we can't use this here, need to make it ourselves a different way, use an STL library, or take from a non-GPL project